lib/queuePopulator/QueuePopulatorExtension.js

```diff
@@ -73,6 +78,34 @@ class QueuePopulatorExtension {
     endBatch() {
         this._batch = null;
     }
+
+    /**
+     * Get currently stored metrics and reset the counters
+     * @return {Object} metrics accumulated since last called
+     */
+    getAndResetMetrics() {
+        const tempStore = Object.assign({}, this._metricsStore);
```

Comment:
Here just do a simple assignment `tempStore = this._metricsStore`, no need to copy.

Reply:
very cool, I thought I had to dupe the object because it would hold reference to the original object